datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1490,6 +1491,92 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_numeric_string_coercion_with_utf8view() {
+        assert_eq!(
```

Comment:
Can you find a test case that fail if the change is reverted instead of the rust test here?

Reply:
There are no test cases that will fail if changes are reverted other than this one.

Reply:
so it means the change is not solving the problem 🤔

Reply:
I'm a bit confused, the function is used [here](https://github.com/apache/datafusion/blob/7ebd993fe47e78006bad33256160716f73b13cef/datafusion/expr-common/src/type_coercion/binary.rs#L532). If reverted, the change will fail the current added test but no other tests. This means that the changes allow support for `Utf8View` in `numeric_string_conversion`

Reply:
If I understand correctly, your test is unit test, it failed if the change reverted but it doesn't mean the change effects the actual query (I guess after we change the signature for most of the function, numeric string coercion might be used at all). I suggest we add end2end test (slt) instead of unit test

Reply:
Ideally this query should pass after your change, but it didn't
```
SELECT make_array(arrow_cast('1.2', 'Utf8View'), 2.3);
```